tests/tensor_ops/test_getitem.py

```diff
@@ -48,6 +47,19 @@ def adv_index_bool_wrap(*arrs):
     return hnp.arrays(shape=arrs[0].shape, dtype=bool)
 
 
+def arb_index_wrap(*arrs):
+    return arbitrary_indices(arrs[0].shape)
+
+
+def test_index_empty():
+    a = Tensor([])
+    assert a[[]].shape == (0,)
+
+    a.sum().backward()
+    # since a is empty, a.grad should be empty and same shape
+    assert np.allclose(a, a.grad)
```

Comment:
`np.allclose` isn't strict about shapes with empty arrays

```python
>>> np.allclose(np.array([[]]) , np.array([]))
True
```
Let's check the shapes explicitly

```suggestion
def test_index_empty():
    a = mg.Tensor([])
    b = a[[]]
    
    assert b.shape == (0,)

    b.sum().backward()
    # since a is empty, a.grad should be empty and same shape
    assert a.shape == a.grad.shape
    assert b.shape == b.grad.shape
```